v3/tasks/20250313-v3-quantamm/test/task.fork.ts

```diff
@@ -0,0 +1,76 @@
+import hre from 'hardhat';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+import { describeForkTest, getForkedNetwork, Task, TaskMode } from '@src';
+import * as expectEvent from '@helpers/expectEvent';
+import { createPoolParams, CreationNewPoolParams, QuantAMMDeploymentInputParams } from '../input';
+
+describeForkTest('QuantAMMPool', 'mainnet', 21818600, function () {
+    let task: Task;
+    let factory: Contract, pool: Contract;
+    let input: QuantAMMDeploymentInputParams;
+    let params: CreationNewPoolParams;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+
+    let accounts:string[];
+    let sender:string;
+
+    const TASK_NAME = '20250313-v3-quantamm';
+    const POOL_CONTRACT_NAME = 'QuantAMMWeightedPool';
+    const FACTORY_CONTRACT_NAME = POOL_CONTRACT_NAME + 'Factory';
+    
+    before('run task', async () => {
+      accounts = await hre.ethers.getSigners() as string[];
+      sender = accounts[0];
+      task = new Task(TASK_NAME, TaskMode.TEST, getForkedNetwork(hre));
+      await task.run({ force: true });
+      factory = await task.deployedInstance(FACTORY_CONTRACT_NAME);
+    });
+
+    before('setup contracts and parameters', async () => {
+      //TODO do we initiate a complete stack or do we take addreses from the index deploy
```

Comment:
The `factory` here, obtained through `task.deployedInstance("name")`, is the address of the QuantAMMWeightedPoolFactory as just deployed in the fork. So, that is taking it from the "index deploy".